File: backend/app/services/categoria_service.py
from sqlalchemy.orm import Session

from app.repositories.categoria_repository import (
    CategoriaRepository,
)
from app.schemas.categoria import CategoriaResponse


class CategoriaService:

    @staticmethod
    def listar(
        db: Session,
    ) -> list[CategoriaResponse]:

        categorias = CategoriaRepository.listar(db)

        return [
            CategoriaResponse.model_validate(categoria)
            for categoria in categorias
        ]

File: frontend/src/modules/catalogos/categorias/services/categoriaService.ts
import api from "@/api/axios";
import type { Categoria } from "../types/categoria";

class CategoriaService {
  async listar(): Promise<Categoria[]> {
    const response = await api.get<Categoria[]>(
      "/categorias",
    );

    return response.data;
  }
}

export default new CategoriaService();